client/my-sites/marketing/connections/service.jsx

```diff
@@ -459,7 +460,12 @@ export class SharingService extends Component {
 	}
 
 	shouldBeExpanded( status ) {
-		if ( this.isMailchimpService() && this.state.justConnected ) {
+		if (
+			this.isMailchimpService() &&
+			( this.state.justConnected ||
+				( this.props.queryArgs && 'mailchimp' in this.props.queryArgs ) )
+		) {
+			window.scrollTo( 0, 9999 );
```

Comment:
It feels little funky to set scroll position inside a function that's just returning true/false. Is there better alternative place for it?

Reply:
I agree it's a bit odd, though I'm not too sure if there's a perfect place for it. I've moved it a bit higher up the file. :)

Reply:
The element might get moved around or new things added under it and it wouldn't be the lowest thing on the screen. That's why we need to scroll to the specific element.

This helper could work here: https://github.com/Automattic/wp-calypso/blob/7a05c7a8a721618405c04cfa20959adf13d7277e/client/lib/scroll-into-viewport/index.js#L55

...perhaps even just `#mailchimp` in the URL works? :-)

Reply:
It might be best to have separate function `scrollToMailchimp` or at least code comment explaining it? It should be obvious when reading code why / what it does.

Reply:
Oh I saw just now that you added code comment. All good! :-) https://github.com/Automattic/wp-calypso/pull/37216/commits/191a2ba4e6ad85929a6e0cfbfbdc405f88bfeb85

Reply:
> ...perhaps even just #mailchimp in the URL works?

I tried adding an anchor but no success :(

> This helper could work here:

I'm struggling to get it working tbh, and nor can I find any decent examples that are applicable to this scenario. This was the closest, but I wasn't sure how to get `querySelector` working: 

https://github.com/Automattic/wp-calypso/blob/59415871b27f65fe1f8c214da3be8fd4383c3aeb/client/components/site-selector/index.jsx#L116

Are there any other examples you're aware of?

Reply:
I think I've got it working now so that is scrolls to the Mailchimp setting instead of the bottom of the screen. :)